Added 3 methods in the backend, fixing testimonials , adding Dockerization and PM2 run

salahawad opened the pull request:
added 3 methods in the backend
/api/company/countries for getting all current system company's locations
/api/country/cities for getting specific country cities
/api/cities/list for getting all cities
/api/countries/list for getting all countries

salahawad committed:
added 3 methods in the backend

added 3 methods in the backend
/api/company/countries for getting all current system company's locations
/api/country/cities for getting specific country cities
/api/cities/list for getting all cities
/api/countries/list for getting all countries

salahawad committed:
Fixing multiple issues

Remove duplicate chatting rooms for same company when applying multiple jobs

Adding deactivate use and company from the backend

adding city field in the jobs

updating some dependencies

salahawad committed:
Adding Dockerization and PM2 run

Adding Dockerization and PM2
Adding dev dependencies for testing and linting

salahawad committed:
Adding missing dependency

Adding osenv dependency

thelebdev commented on app/models/Company.js:
```diff
@@ -169,47 +172,45 @@ schema.statics.updateView = async function (id) {
-    console.log("comparePassword ,,,,", password, this.password)
-    // header already sent
-    return await bcrypt.compare(password, this.password);
+  console.log("comparePassword ,,,,", password, this.password);
```
We should remove any log or mention of a user's password

thelebdev commented on app/models/Company.js:
```diff
@@ -169,47 +172,45 @@ schema.statics.updateView = async function (id) {
-    return fullpicture;
+schema.virtual("fullpicture").get(function () {
+  let fullpicture = "";
+  if (this.picture.length > 2) {
```
I'm not sure about the full functionality of this function, but if no sanitation is implemented, we should limit the type of files that are uploadable and the size of the file (dimensions + size) from the backend to avoid cross-site scripting and files large enough to break the system

thelebdev commented on app/models/User.js:
```diff
@@ -1,264 +1,258 @@
-    console.log("comparePassword ,,,,", password, this.password)
-    // header already sent
-    return await bcrypt.compare(password, this.password);
+  console.log("comparePassword ,,,,", password, this.password);
```
Let's remove any mentions of the user's password

salahawad committed:
Adding logger, removing some console logging, removing auth for some apis

Adding winston logger and configuring log files
Removing some console logging where it used to log sensitive data
Removing auth for getcompanies and get users for paging

salahawad committed:
unsubmitted

salahawad commented:
Merging into develop branch
